File: backend/src/main/kotlin/ch/streckeisen/mycv/backend/account/dto/AuthResponseDto.kt
package ch.streckeisen.mycv.backend.account.dto

import ch.streckeisen.mycv.backend.account.AccountStatus
import ch.streckeisen.mycv.backend.cv.profile.ThumbnailDto
import com.fasterxml.jackson.annotation.JsonProperty

data class AuthResponseDto(
    val username: String,
    val authLevel: AccountStatus,
    val displayName: String?,
    val language: String?,
    @get:JsonProperty(value = "hasProfile")
    val hasProfile: Boolean,
    val thumbnail: ThumbnailDto?
)

File: frontend/src/types/account/AuthResponseDto.ts
import { AuthLevel } from '@/types/account/AuthLevel.ts'
import { ThumbnailDto } from '@/types/account/ThumbnailDto.ts'

export interface AuthResponseDto {
  username: string
  displayName: string
  authLevel: AuthLevel
  language: undefined | string,
  hasProfile: boolean,
  thumbnail: ThumbnailDto | undefined
}